Framework/Geometry/src/RandomPoint.cpp

```diff
@@ -100,8 +105,8 @@ Kernel::V3D inHollowCylinder(const detail::ShapeInfo &shapeInfo,
   const double c2 = geometry.radius * geometry.radius;
   const double r{std::sqrt(c1 + (c2 - c1) * r2)};
   const double z{geometry.height * r3};
+  const Kernel::V3D basis1{geometry.axis};
   const Kernel::V3D alongAxis{geometry.axis * z};
-  const Kernel::V3D &basis1{geometry.axis};
   auto localPoint = localPointInCylinder(basis1, alongAxis, polar, r);
```

Comment:
Can we call `localPointInCylinder` with `geometry.axis` directly here to avoid the temporary?